pages/_document.tsx

```diff
@@ -16,6 +19,10 @@ export default class MyDocument extends Document {
     const sheet = new ServerStyleSheet();
     const originalRenderPage = ctx.renderPage;
 
+    if (ctx?.req?.headers?.host) {
+      host = `https://${ctx?.req?.headers?.host}`;
```

Comment:
this can cause minor but hard-to-debug issues if someone will attempt to request us with different host. please make it one-time assignment or define in env